### PR TITLE
add test, improve README and remove duplicate logger creation

### LINKED ISSUE #17: Write tests with mocks

### PR DESCRIPTION
**Pull request by xkal36**
Closes #17, #21 and #18

#### Checklist

- [x ] run `npm run test` and `npm run benchmark`
- [x ] tests and/or benchmarks are included
- [x ] documentation is changed or added
- [x ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    data: { ...mockIssuesData, items: [mockIssue1] }
+  })
+  searchIssuesStub.onCall(1).returns({
+    ...mockSearchIssuesAndPullRequests,
+    data: { ...mockIssuesData, items: [mockIssue2] }
+  })
+
+  const build = t.mock('./app', {
+    './fetchIssues': {
+      fetchIssues,
```
Comment (xkal36) [inline test.js:107]: Not too happy with how I have done this - tap seems to be mocking the whole module and so when I removed this line, it couldn't find the `fetchIssues` function. Not how I understood this should work. Open to suggestions on how to do this in a cleaner way

**Review by mcollina — COMMENTED** [review]
lgtm
